### PR TITLE
BUG: Correct read_stata

### PR DESCRIPTION
**Pull request by bashtage**
Update and silence for future warnings from pandas/main

### CODE REVIEW

File: tests/test_frame.py [hunk 1]
```diff
@@ -1505,27 +1504,31 @@ def test_frame_scalars_slice() -> None:
 
     # Note: bool_ cannot be tested since the index is object and pandas does not
     # support boolean access using loc except when the index is boolean
-    check(assert_type(df.loc[str_], pd.Series), pd.Series)
+    with pytest.warns(FutureWarning, match="Comparison of Timestamp with datetime"):
```
Comment (twoertwein) [inline tests/test_frame.py:1507]: If we want to treat deprecation/future-warnings on main the same we do for the released pandas versions, we could change the type annotations to make mypy/pyright fail in these cases and make them negative tests.

File: pandas-stubs/core/frame.pyi [hunk 2]
```diff
@@ -633,37 +633,6 @@ class DataFrame(NDFrame, OpsMixin):
         axis: AxisType = ...,
         fill_value: Hashable | None = ...,
     ) -> DataFrame: ...
-    @overload
-    def set_index(
```
Comment (twoertwein) [inline pandas-stubs/core/frame.pyi:637]: I think this should wait until 1.5 is released. 1.5 deprecates `inplace` and adds the new `copy` argument.

Might be worth writing a `@skip_on_nightly` decorator for the tests.

Comment (Dr-Irv) [reply pandas-stubs/core/frame.pyi:671]: Other option is we start maintaining a fork related to 1.5 changes.

**Review by twoertwein — APPROVED** [review]
`set_index` isn't ideal, but letting it always return a `DataFrame` is consistent with how it will be in the near future.

**Comment by Dr-Irv** [comment]
I'm holding off on merging this until 1.5 is released.

**Comment by twoertwein** [comment]
Definitely good to wait for 1.5 - especially since the `set_index` behavior might get reverted before 1.5 releases https://github.com/pandas-dev/pandas/issues/48141#issuecomment-1238505678

**Comment by twoertwein** [comment]
Since https://github.com/pandas-dev/pandas/pull/48417 is merged, we can remove the `set_index` changes and probably merge this PR before 1.5 releases as it hopefully doesn't disrupt 1.4.x.

File: pandas-stubs/core/frame.pyi (hunk 2 above)
Comment (bashtage) [reply pandas-stubs/core/frame.pyi]: I reverted this since inplace is back for 1.5.0

File: tests/test_frame.py (hunk 1 above)
Comment (bashtage) [reply tests/test_frame.py]: I agree with this.  We should go through the FutureWarnings after the next release and remove problematic code.

**Comment by bashtage** [comment]
Doesn't do much any more, but does fix a bug in read_stata

**Review by Dr-Irv — APPROVED** [review]

**Comment by Dr-Irv** [comment]
thanks @bashtage

**Pull request merged**